cli/ceph/nfs/export/qos.py

```diff
@@ -25,26 +25,69 @@ def get(self, nfs_name, export, **kwargs):
-        max_export_read_bw : required  - pass in kwargs
+        self,
+        qos_type: str,
+        nfs_name: str,
+        export: str,
+        operation: str = "bandwidth_control",
+        **kwargs,
+    ) -> str:
+        """Enable QoS controls for an NFS cluster.
+
+        Args:
+            cluster_id: NFS cluster identifier
+            qos_type: QoS type (PerShare/PerClient/PerShare-PerClient)
+            operation: Operation type (default: bandwidth_control)
+            **kwargs: Required parameters based on qos_type:
+                - PerShare: Either (max_export_write_bw + max_export_read_bw)
+                            OR max_export_combined_bw
+                - PerClient: Either (max_client_write_bw + max_client_read_bw)
+                             OR max_client_combined_bw
+                - PerShare-PerClient: Combination of PerShare AND PerClient requirements
+
+        Returns:
+            Command execution output
+
+        Raises:
+            ValueError: For missing/invalid parameters or qos_type
         """
-        if "max_export_write_bw" not in kwargs or "max_export_read_bw" not in kwargs:
-            raise Exception(
-                'For "PerShare" - kwargs "max_export_write_bw" and "max_export_read_bw" required'
-                + ', example - max_export_write_bw = "10MB"'
-            )
+        # Validate qos_type
+        params = []
 
-        cmd = f"{self.base_cmd} enable {operation} {nfs_name} {export} {build_cmd_from_args(**kwargs)}"
-        out = self.execute(sudo=True, cmd=cmd)
-        if isinstance(out, tuple):
-            return out[0].strip()
-        return out
+        # Handle Export (PerShare) parameters
+        if qos_type in ["PerShare", "PerShare_PerClient"]:
+            if "max_export_combined_bw" in kwargs:
+                params.append(
+                    f"--max_export_combined_bw {kwargs['max_export_combined_bw']}"
+                )
+            else:
+                params.extend(
+                    [
+                        f"--max_export_write_bw {kwargs['max_export_write_bw']}",
+                        f"--max_export_read_bw {kwargs['max_export_read_bw']}",
+                    ]
+                )
+
+        # Handle Client (PerClient) parameters
+        if qos_type in ["PerShare_PerClient"]:
+            if "max_client_combined_bw" in kwargs:
+                params.append(
+                    f"--max_client_combined_bw {kwargs['max_client_combined_bw']}"
+                )
+            else:
+                params.extend(
+                    [
+                        f"--max_client_write_bw {kwargs['max_client_write_bw']}",
+                        f"--max_client_read_bw {kwargs['max_client_read_bw']}",
+                    ]
+                )
+        cmd = (
+            f"{self.base_cmd} enable {operation} {nfs_name} {export} "
+            f"{' '.join(params)}"
+        )
+        result = self.execute(sudo=True, cmd=cmd)
```

Comment:
It is okay to write multiple methods. IMHO, we should ensure the required `kwargs` are provided as it is made generic.

Multiple methods help us avoid the need to have multiple `if statements`